### DISCUSSION TITLE
Home Assistant reporting

### DISCUSSION THREAD

**Discussion by ScratMan** [zwave-js/zwave-js-ui#1939] [General]
Hello,

After starting Home Assistant, the log shows a warning about measurement unit for the cumulated rain sensor of the Popp Z-Rain sensor (POPE700168, ID 0x0154-0x0004-0x0011).

> Logger: homeassistant.components.sensor.recorder
> Source: components/sensor/recorder.py:269
> Integration: Capteur (documentation, issues)
> First occurred: 12:20:10 (1 occurrences)
> Last logged: 12:20:10
> 
> sensor.popp_rain_sensor_water_consumed_cubic_meters has unknown unit Cubic meters

In HA the **sensor.popp_rain_sensor_water_consumed_cubic_meters** gets the following attributes automatically

```state_class: total_increasing
meter_type: 3
meter_type_name: WATER
unit_of_measurement: Cubic meters
friendly_name: 'Popp Rain Sensor: Water Consumed [Cubic meters]'
device_class: energy
```

I looked into the [configuration file](https://github.com/zwave-js/node-zwave-js/blob/dc0986bff07c3b784d9492f35f1440ad852cd0d3/packages/config/config/devices/0x0154/pope700168.json), but there is no mention of cubic meters.
I can't find where this _Cubic meters_ unit is defined. Is it defined by ZWave-JS ?

Thanks

**Comment by robertsLando** [comment]
This is something related to hass integration. I have no clue. Try to open an issue on hass repo :)

**Comment by kpine** [comment]
The unit is defined by [node-zwave-js](https://github.com/zwave-js/node-zwave-js/blob/fe4ba0282a0331f0052f8cfbf8748bb65ec8ae09/packages/config/config/meters.json#L29).

The meter entity has been assigned device class `energy`, but HA does not have `Cubic meters` unit for energy. The only [valid units for energy](https://github.com/home-assistant/core/blob/a989fd2e663472c3c69f1fed3a8446f21d523bd0/homeassistant/components/sensor/recorder.py#L99-L103) are `Wh`, `kWh`, and `MWH`. There is no device class for water.

There is a device class `gas`, and it has a unit `m³`. You could [customize](https://www.home-assistant.io/docs/configuration/customizing-devices/#customizing-entities) the entity with these values as a workaround. The actual fix would involve a change to the zwave-js integration to support the unit `m³` properly. Maybe a new device class is necessary.

A similar issue has been reported [here](https://github.com/home-assistant/core/issues/59386).
